src/semantic-router/pkg/config/config.go

```diff
@@ -264,6 +268,73 @@ type Category struct {
 	ModelScores          []ModelScore `yaml:"model_scores"`
 }
 
+// ModelReasoningSyntax defines how a model handles reasoning mode
+type ModelReasoningSyntax struct {
+	Type      string `yaml:"type"`      // "chat_template_kwargs" or "reasoning_effort"
+	Parameter string `yaml:"parameter"` // "thinking", "enable_thinking", "reasoning_effort", etc.
+}
+
+// ModelReasoningConfig defines reasoning configuration for a model family
+type ModelReasoningConfig struct {
+	Name            string               `yaml:"name"`
+	Patterns        []string             `yaml:"patterns"` // patterns to match against model names
+	ReasoningSyntax ModelReasoningSyntax `yaml:"reasoning_syntax"`
+}
+
+// FindModelReasoningConfig finds the appropriate reasoning configuration for a given model name
+func (rc *RouterConfig) FindModelReasoningConfig(modelName string) *ModelReasoningConfig {
+	if rc == nil || len(rc.ModelReasoningConfigs) == 0 {
+		return nil
+	}
+
+	modelLower := strings.ToLower(strings.TrimSpace(modelName))
+
+	// Look for exact pattern matches first
+	for _, config := range rc.ModelReasoningConfigs {
+		for _, pattern := range config.Patterns {
+			if matchesPattern(modelLower, strings.ToLower(pattern)) {
+				return &config
+			}
+		}
+	}
+
+	// Look for a default/fallback configuration
+	for _, config := range rc.ModelReasoningConfigs {
+		if config.Name == "default" || contains(config.Patterns, "*") {
```

Comment:
I think it's better not to hardcode config.Name and just keep the judgment logic for config.Patterns. 

Otherwise, we will always need to consider the default name separately in our subsequent processing logic.

WDYT?